### ISSUE TITLE
calculation=get_pchg failed

### ISSUE THREAD

**Issue by sgyang345** [deepmodeling/abacus-develop#2923]
### Describe the bug

**When running ABACUS 3.3.3 with `calculation=get_pchg` , the program failed.**

The logs are as follows:
 ---------------------------------------------------------
...
 SPIN    KPOINTS         PROCESSORS  NBASE
 1       Gamma           16          113
 Use Systematically Improvable Atomic bases

 ELEMENT ORBITALS        NBASE       NATOM       XC
 Mo      4s2p2d1f-10au   27          1
 Se      2s2p1d-10au     13          4
 W       4s2p2d2f-10au   34          1

 Initial plane wave basis and FFT box

 Read in gamma point wave function files
 name is = OUT.ABACUS/LOWF_GAMMA_S1.dat

_abacus: /opt/software/ABACUS/abacus-develop-v3.3.3/source/module_io/read_wfc_nao.cpp:282: int ModuleIO::read_wfc_nao(double **, const int &, const Parallel_Orbitals *, psi::Psi<double, psi::DEVICE_CPU> *, elecstate::ElecState *): Assertion `(i+1)==ib' failed._

=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 4482 RUNNING AT 69
=   KILLED BY SIGNAL: 6 (Aborted)

### Additional Context

The test example file is as follows:

[abacus_get_pchg.zip](https://github.com/deepmodeling/abacus-develop/files/12560305/abacus_get_pchg.zip)

### Task list for Issue attackers

- [x] Verify the issue is not a duplicate.
- [x] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [x] Error message.
- [ ] Environment details.
- [x] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by AsTonyshment** [comment]
Apologies for the delayed response. After some troubleshooting, I've identified the error resulting from a mismatch between the wavefunction coefficient file `LOWF_GAMMA_S1` provided by you and the system you've computed. Specifically, for the two-dimensional material with 1 Mo, 4 Se, and 1 W as per your computation, the SCF output log `running_scf.log` indicates an orbital count of `NLOCAL = 113`. However, the `LOWF_GAMMA_S1` you provided lists the orbital count as 791 in the second line. ABACUS loops over `NLOCAL` while reading the wavefunction coefficient file, and the program exits with an error if the coefficient count does not match the orbital count.

Furthermore, I noticed `113 (number of orbitals)` at lines 45040 and 46160, indicating that you did not delete the previous OUT.ABACUS directory before initiating a new computation. This resulted in the accumulation of multiple computation results into the wavefunction coefficient file `LOWF_GAMMA_S1`. I recommend checking for potential interference from previous computations before initiating each new computation.

In conclusion, this issue is not a bug within the program itself. This issue will be closed.
